.buildkite/scripts/package.sh

```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env bash
+##
+##  Builds the distribution packages: tarballs, RPMs, Docker images, etc.
+##  It runs for both linux/amd64 and linux/arm64. On linux/amd64 it builds all packages;
+##  on linux/arm64 it builds only Docker images.
+##
+
+set -eo pipefail
+
+TYPE="$1"
+PLATFORM_TYPE=$(uname -m)
+
+PLATFORMS=''
+PACKAGES=''
+if [[ ${PLATFORM_TYPE} == "arm" || ${PLATFORM_TYPE} == "aarch64" ]]; then
+  PLATFORMS='linux/arm64'
+  PACKAGES='docker'
+fi
+
+MAKE_GOAL=release-manager-snapshot
+if [[ ${TYPE} == "staging" ]]; then
+  MAKE_GOAL="release-manager-release"
+fi
+
+# Prepare the context for using a different workspace
+# so the package can use the right folder location.
+# Buildkite uses the pipeline name for the workspace and
+# it breaks the package build process.
+cp -rf . ../apm-server
```

Comment:
This is an alternative to use a different location so the Beats build system can work otherwise the workspace will contain a name which does not match `apm-server` and will fail with:

```
Generated fields.yml for apm-server-package to /opt/buildkite-agent/builds/bk-agent-prod-gcp-1692623868511889735/elastic/apm-server-package/build/fields/fields.yml
--
  | Cannot collect fields.yml files: open /opt/buildkite-agent/builds/bk-agent-prod-gcp-1692623868511889735/elastic/apm-server-package/x-pack/apm-server-package: no such file or directory
```

`elastic/apm-server-package` should be `elastic/apm-server-package`

Reply:
Unfortunately in Buildkite is required to either use a plugin or a hook in the Buildkite agent.

I'd say we can keep this approach since it's only required for the `7.17` branch.